src/com/amazon/ion/impl/bin/IonRawBinaryWriter.java

```diff
@@ -1491,83 +1488,60 @@ public void writeBytes(byte[] data, int offset, int length) throws IOException
         }
     }
 
-    public void flush() throws IOException {}
+    public void flush() throws IOException {
+        if (!closed) {
```

Comment:
Why has this behavior moved in to `flush`?

Reply:
Because it is the flushing logic. It was only in finish because flush was a no op before.

Reply:
Because finish used have flush's responsibilities and flush used to be a no op function